portal/src/helpers/domain_parsing.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { parseDomain, ParseResultType } from "parse-domain";
+import { Path } from "../types/index";
+
+/**
+ * Returns the domain (e.g. "example.com") of the given URL.
+ * @param orig_url The URL to extract the domain from. e.g. "https://example.com"
+ * @returns The domain of the URL. e.g. "example.com"
+ */
+export function getDomain(url: URL): string {
+    const parsed = parseDomain(url.hostname);
+    if (parsed.type === ParseResultType.Listed) {
+        const domain = parsed.domain + "." + parsed.topLevelDomains.join(".");
+        return domain;
+    } else if (parsed.type === ParseResultType.Reserved) {
+        return parsed.labels[parsed.labels.length - 1];
+    } else {
+        console.error("Error while parsing domain name:", parsed);
+        throw new Error("Error while parsing domain name");
+    }
+}
+
+/**
+* Given a URL, returns the subdomain and path.
+* @param url e.g. "https://subname.name.walrus.site/"
+* @returns Path object e.g. { subdomain: "subname.name", path: "/index.html"}
+*/
+
+export function getSubdomainAndPath(url: URL): Path | null {
+    const parsed = parseDomain(url.hostname);
+    if (parsed.type === ParseResultType.Listed) {
+        return {
+            subdomain: parsed.subDomains.join("."),
+            path: url.pathname == "/" ? "/index.html" : removeLastSlash(url.pathname)
+        } as Path;
+    } else if ( parsed.type === ParseResultType.Reserved) {
+        return {
+            subdomain: parsed.labels.slice(0, parsed.labels.length-1).join('.'),
+            path: url.pathname == "/" ? "/index.html" : removeLastSlash(url.pathname)
+        } as Path;
+    }
+    return null;
+}
```

Comment:
Yes I considered doing this but I ended up preferring two functions instead of one, because there is better separation of concerns. Even though we repeat a little bit of code, it's a good trade-off for readability imho. 

If you disagree, I can merge them into one, no problem. 👍

Reply:
My concern is that when we will need to make the change in #124 we will need to remember to touch both of these functions. Therefore, I would be in favor of not duplicating the logic.

If you like the two top level functions, maybe we can keep them and have them call internally another `splitUrl` functions that returns `{subdomain: "subsub.sub", domain: "example.com", path: "/some/path.html"}`? Such that the URL processing logic is as centralized as possible.